pyfar/io/io.py

```diff
@@ -461,6 +463,320 @@ def default_audio_subtype(format):
+                if any(idxes):
+                    data_out[:, i_expression, parameters_idx, i_domain] \
+                        = data_expressions[:, idxes].flatten()
+                else:
+                    data_inconsistent = True
+                    warnings.warn(r'Parameter data is inconsistent. Missing \
+                        data is filled with nans.')
+
+    # reshape data to final shape
+    if not data_inconsistent:
+        data_out = np.reshape(data_out, new_shape)
+
+    # create object
+    comment = ', '.join(
+        ' '.join(x) for x in zip(all_expressions, units))
+    if domain == 'freq':
+        data = FrequencyData(
+            data_out, domain_data, dtype=dtype,
+            comment=comment)
+    else:
+        data = TimeData(
+            data_out, domain_data, comment=comment)
+
+    # create coordinates
+    if n_dimension > 0:
+        coords_data = raw_data[:, 0:n_dimension].astype(np.float)
+        x = coords_data[:, 0]
+        y = coords_data[:, 1] if coords_data.shape[1] > 1 else np.zeros(
+            coords_data[:, 0].shape)
+        z = coords_data[:, 2] if coords_data.shape[1] > 2 else np.zeros(
+            coords_data[:, 0].shape)
+        coordinates = Coordinates(x, y, z)
+        return data, coordinates
+    return data, None
+
+
+def read_comsol_header(filename):
+    """Returns header information from exported data from COMSOL.
+
+    Parameters
+    ----------
+    filename : str, Path
+        Input file. Excepted input files are '.txt', '.dat' and '.csv'. '.csv'
+        Files are strongly recommended, since .txt and .dat are inconvenient
+        in their format.
+
+    Returns
+    -------
+    expressions : list[str]
+        This list of expressions defines the output expressions. An example
+        can be like ``expressions=['pabe.p_t']``. Also see
+        :py:func:`~pyfar.dsp.comsol_read`.
+    expressions_unit : list[str]
+        This list of expressions unit defines the units of the output
+        expressions. An example can be like ``expressions_unit=['Pa']``.
+    parameters : dict
+        This dict contains all parameters ins from the input file. If no
+        parameters are available, an emtpy dict is returnd. An example can be
+        like ``parameters={'theta': [0.0, 0.7854], 'phi': [0., 1.5708]}``.
+    domain : string
+        Returns the domain of the input data. Note that ``'lambda'`` domain
+        data is not allowed.
+
+        ``'freq'``
+            The data are in frequency domain.
+        ``'time'``
+            The data are in time domain.
+    domain_data : list[]
+        This list contains the domain data from the input file. Depending
+        on the data in the input file, the output will be float or complex.
+
+    Raises
+    ------
+    SyntaxError
+        Raises SyntaxError if the file format is not supported.
+    ValueError
+        Raises ValueError if the domain is not supported.
+
+    Examples
+    --------
+    Returns all expressions
```

Comment:
Should we include a print of the output in the example? Otherwise the example would only be a function call.